blueprints/snapshots.and.monitoring.md

```diff
@@ -0,0 +1,828 @@
+		"healthPollingInterval": 6000,
+		"heartbeatPollingInterval": 3000,
+		"peerPollingInterval": 3000
+	},
+	"deliveryServices": {
+		"demo1": {
+			"globalMaxTps": 0,
+			"globalMaxMbps": 0
+		}
+	},
+	"profiles": {
+		"ATS_MID_TIER_CACHE": {
+			"parameters": {
+				"health.connection.timeout": "2000",
+				"health.polling.url": "http://${hostname}/_astats?application=&inf.name=${interface_name}",
+				"health.polling.format": "",
+				"health.polling.type": "",
+				"history.count": "30",
+				"MinFreeKbps": "0"
+			},
+			"thresholds": {
+				"health_threshold": {
+					"availableBandwidthInKbps": {
+						"value": 1750000,
+						"comparator": ">"
+					},
+					"loadavg": {
+						"value": 25,
+						"comparator": "<"
+					},
+					"queryTime": {
+						"value": 1000,
+						"comparator": "<"
+					}
+				}
+			}
+		},
+		"ATS_EDGE_TIER_CACHE": {
+			"parameters": {
+				"health.connection.timeout": "2000",
+				"health.polling.url": "http://${hostname}/_astats?application=&inf.name=${interface_name}",
+				"health.polling.format": "",
+				"health.polling.type": "",
+				"history.count": "30",
+				"MinFreeKbps": "0"
+			},
+			"thresholds": {
+				"health_threshold": {
+					"availableBandwidthInKbps": {
+						"value": 1750000,
+						"comparator": ">"
+					},
+					"loadavg": {
+						"value": 25,
+						"comparator": "<"
+					},
+					"queryTime": {
+						"value": 1000,
+						"comparator": "<"
+					}
+				}
+			}
+		}
+	},
+	"topologies": {
+		"my-topology": {
+			"nodes": ["CDN_in_a_Box_Edge"]
+		}
+	},
+	"trafficMonitors": {
+		"trafficmonitor": {
+			"fqdn": "trafficmonitor.infra.ciab.test",
+			"ipv4Address": "172.20.0.13",
+			"ipv6Address": null,
+			"port": 80,
+			"profile": "RASCAL-Traffic_Monitor",
+			"status": "ONLINE"
+		}
+	}
+}}
+```
+
+### CDN Snapshots
+```typescript
+interface MaxmindDefaultOverride {
+	countryCode: string;
+	latitude: number;
+	longitude: number;
+}
+
+// All of these properties used to be strings.
+interface SOA {
+	admin: string;
+	expire: number;//integer
+	minimum: number;//integer
+	refresh: number;//integer
+	retry: number;//integer
+}
+
+interface DNSBypassDestination {
+	cname: string | null;
+	ipv4Address: string | null;
+	ipv6Address: string | null;
+	ttl: number;//integer
+}
+
+interface HTTPBypassDestination {
+	fqdn: string;
+	port: number;//integer
+}
+
+interface DNSMatchItem {
+	protocol: "DNS";
+	// used to be named "matchlist"
+	matchList: Array<{
+		regex: string,
+		// used to be named "match-type"
+		type: "HOST"
+	}>;
+}
+
+interface HTTPMatchItem {
+	protocol: "HTTP";
+	// used to be named "matchlist"
+	matchList: Array<{
+		regex: string,
+		// used to be named "match-type"
+		type: "HOST" | "PATH" | "HEADER"
+	}>;
+}
+
+interface BaseDeliveryService {
+	// used to be a string containing a boolean, instead of just a boolean.
+	anonymousBlockingEnabled: boolean;
+	// Formerly 'ttl' - now consistent with the DS property it represents.
+	ccrDNSTTL: null | number;//integer
```

Comment:
rename this to get rid of ccr?

Reply:
I'd love to- but not until that can be done on Delivery Services themselves. This is the same name as the property on the DS and I think that consistency is more important than a good name, because it makes it very obvious where that data came from.